src/host-config.ts

```diff
@@ -214,15 +215,30 @@ export function get(): Bluebird<HostConfig> {
 	});
 }
 
-export function patch(conf: HostConfig): Bluebird<void> {
-	const promises: Array<Promise<void>> = [];
-	if (conf != null && conf.network != null) {
-		if (conf.network.proxy != null) {
-			promises.push(setProxy(conf.network.proxy));
-		}
-		if (conf.network.hostname != null) {
-			promises.push(setHostname(conf.network.hostname));
-		}
+export async function patch(conf: HostConfig, force: boolean): Promise<void> {
+	const appId = await config.get('applicationId');
+	if (!appId) {
+		throw new InternalInconsistencyError('Could not find an appId');
 	}
-	return Bluebird.all(promises).return();
+
+	const skipLock =
+		(
+			await Bluebird.any([
+				dbus.servicePartOf('balena-hostname'),
+				dbus.servicePartOf('resin-hostname'),
+			])
+		).includes('config-json.target') === false;
```

Comment:
I would remove this, I don't see much of an advantage in adding this extra complexity. The API from this PR always uses `updateLock`, no matter the OS version.